### PR TITLE
test: fix e2e changeover test

### PR DESCRIPTION
**Pull request by bermuell**
## Description
---

Fix e2e changeover tests
- uses now MsgUpgradeProposal instead of LegacyUpgradeProposal
- fixes different issues breaking the existing tests

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code 

## Summary by CodeRabbit

- **New Features**
	- Enhanced governance capabilities with the addition of new configuration parameters for proposals and IBC upgrades.
	- Improved lifecycle management in the application initialization process.

- **Bug Fixes**
	- Refined upgrade proposal handling to ensure correct processing without affecting functionality.

- **Refactor**
	- Updated functions to replace deprecated action types, improving clarity and maintainability.

- **Chores**
	- Simplified a shell script command for better readability.

### CODE REVIEW

File: tests/e2e/action_rapid_test.go [hunk 1]
```diff
@@ -163,9 +163,9 @@ func GetStartSovereignChainActionGen() *rapid.Generator[StartSovereignChainActio
 	})
 }
 
-func GetSubmitLegacyUpgradeProposalActionGen() *rapid.Generator[LegacyUpgradeProposalAction] {
-	return rapid.Custom(func(t *rapid.T) LegacyUpgradeProposalAction {
-		return LegacyUpgradeProposalAction{
+func GetSubmitLegacyUpgradeProposalActionGen() *rapid.Generator[UpgradeProposalAction] {
```
Comment (sainoe) [inline tests/e2e/action_rapid_test.go:166]: nit: this function can be renamed to `GetSubmitUpgradeProposalActionGen`, right?

File: tests/e2e/json_utils.go [hunk 2]
```diff
@@ -217,7 +217,7 @@ func UnmarshalMapToActionType(rawAction json.RawMessage, actionTypeString string
 			return a, nil
 		}
 	case "main.LegacyUpgradeProposalAction":
```
Comment (sainoe) [inline tests/e2e/json_utils.go:219]: Same thing here, this case can be renamed to `main.UpgradeProposalAction`.

File: tests/e2e/actions_sovereign_chain.go [hunk 3]
```diff
@@ -103,14 +103,84 @@ func (tr Chain) startSovereignChain(
...
+		log.Fatal(err, "\n", string(bz))
+	}
+	if verbose {
+		log.Println("Response to submit-proposal: ", string(bz))
+	}
+
+	tr.waitBlocks(action.ChainID, 1, 15*time.Second)
+}
+
+func (tr *Chain) submitLegacyUpgradeProposalX(action UpgradeProposalAction, verbose bool) {
```
Comment (sainoe) [inline tests/e2e/actions_sovereign_chain.go:183]: Is this still needed?

File: tests/e2e/state.go [hunk 4]
```diff
@@ -454,6 +457,7 @@ func (tr Commands) GetProposal(chain ChainID, proposal uint) Proposal {
 			},
 		}
 	case "/cosmos.upgrade.v1beta1.SoftwareUpgradeProposal":
```
Comment (sainoe) [inline tests/e2e/state.go:459]: This case can be removed, right?

File: app/consumer/app.go [hunk 5]
```diff
@@ -490,7 +490,6 @@ func New(
 		vestingtypes.ModuleName,
 		ibcconsumertypes.ModuleName,
 	)
-	app.SetPreBlocker(app.PreBlocker)
```
Comment (bermuell) [inline app/consumer/app.go:493]: duplicate to line 561

**Review by MSalopek — APPROVED** [review]
LGTM!

Waiting for tests

**Review by mpoke — APPROVED** [review]

**Pull request merged**